freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/DetailedStackStatus.java

```diff
@@ -1,80 +1,99 @@
+    STOP_REQUESTED(Status.STOP_REQUESTED, false),
+    STOP_IN_PROGRESS(Status.STOP_IN_PROGRESS, false),
+    STOPPED(Status.STOPPED, false),
+    STOP_FAILED(Status.STOP_FAILED, false),
     // Upscale statuses
-    UPSCALE_REQUESTED(Status.UPDATE_REQUESTED),
-    UPSCALE_IN_PROGRESS(Status.UPDATE_IN_PROGRESS),
-    UPSCALE_COMPLETED(Status.AVAILABLE),
-    UPSCALE_FAILED(Status.UPSCALE_FAILED),
+    UPSCALE_REQUESTED(Status.UPDATE_REQUESTED, true),
+    UPSCALE_IN_PROGRESS(Status.UPDATE_IN_PROGRESS, true),
+    UPSCALE_COMPLETED(Status.AVAILABLE, true),
+    UPSCALE_FAILED(Status.UPSCALE_FAILED, false),
     // Downscale statuses
-    DOWNSCALE_REQUESTED(Status.UPDATE_REQUESTED),
-    DOWNSCALE_IN_PROGRESS(Status.UPDATE_IN_PROGRESS),
-    DOWNSCALE_COMPLETED(Status.AVAILABLE),
-    DOWNSCALE_FAILED(Status.DOWNSCALE_FAILED),
+    DOWNSCALE_REQUESTED(Status.UPDATE_REQUESTED, true),
+    DOWNSCALE_IN_PROGRESS(Status.UPDATE_IN_PROGRESS, true),
+    DOWNSCALE_COMPLETED(Status.AVAILABLE, true),
+    DOWNSCALE_FAILED(Status.DOWNSCALE_FAILED, false),
     // Repair statuses
-    REPAIR_REQUESTED(Status.UPDATE_REQUESTED),
-    REPAIR_IN_PROGRESS(Status.UPDATE_IN_PROGRESS),
-    REPAIR_COMPLETED(Status.AVAILABLE),
-    REPAIR_FAILED(Status.REPAIR_FAILED),
+    REPAIR_REQUESTED(Status.UPDATE_REQUESTED, true),
+    REPAIR_IN_PROGRESS(Status.UPDATE_IN_PROGRESS, true),
+    REPAIR_COMPLETED(Status.AVAILABLE, true),
+    REPAIR_FAILED(Status.REPAIR_FAILED, false),
     // Termination statuses
-    DEREGISTERING_WITH_CLUSTERPROXY(Status.DELETE_IN_PROGRESS),
-    DEREGISTERING_CCM_KEY(Status.DELETE_IN_PROGRESS),
-    DELETE_IN_PROGRESS(Status.DELETE_IN_PROGRESS),
-    DELETE_COMPLETED(Status.DELETE_COMPLETED),
-    DELETE_FAILED(Status.DELETE_FAILED),
+    DEREGISTERING_WITH_CLUSTERPROXY(Status.DELETE_IN_PROGRESS, false),
+    DEREGISTERING_CCM_KEY(Status.DELETE_IN_PROGRESS, false),
+    DELETE_IN_PROGRESS(Status.DELETE_IN_PROGRESS, false),
+    DELETE_COMPLETED(Status.DELETE_COMPLETED, false),
+    DELETE_FAILED(Status.DELETE_FAILED, false),
     // Rollback statuses
-    ROLLING_BACK(Status.UPDATE_IN_PROGRESS),
+    ROLLING_BACK(Status.UPDATE_IN_PROGRESS, false),
     // The stack is available
-    AVAILABLE(Status.AVAILABLE),
+    AVAILABLE(Status.AVAILABLE, true),
     // Instance removing status
-    REMOVE_INSTANCE(Status.UPDATE_IN_PROGRESS),
+    REMOVE_INSTANCE(Status.UPDATE_IN_PROGRESS, false),
     // Cluster operation is in progress
-    CLUSTER_OPERATION(Status.UPDATE_IN_PROGRESS),
+    CLUSTER_OPERATION(Status.UPDATE_IN_PROGRESS, true),
     // Wait for sync
-    WAIT_FOR_SYNC(Status.WAIT_FOR_SYNC),
-    UNREACHABLE(Status.UNREACHABLE),
-    DELETED_ON_PROVIDER_SIDE(Status.DELETED_ON_PROVIDER_SIDE),
-    UNHEALTHY(Status.UNHEALTHY);
+    WAIT_FOR_SYNC(Status.WAIT_FOR_SYNC, false),
+    UNREACHABLE(Status.UNREACHABLE, false),
+    DELETED_ON_PROVIDER_SIDE(Status.DELETED_ON_PROVIDER_SIDE, false),
+    UNHEALTHY(Status.UNHEALTHY, true);
```

Comment:
I am definitely missing something but figured I'd ask, why is it available when it is unhealthy?

Reply:
Let's say we are running FreeIPA HA and there are 3 instances. One is in a bad state and two are responding correctly. Then the cluster would be unhealthy but that means that someone needs to repair it but FreeIPA is able to respond to all requests in this case. It definitely isn't perfect. For example, if all nodes are unhealthy then it is possible FreeIPA is unable to respond. But usually, when all nodes are unhealthy we see that FreeIPA is unreachable, stopped, failed, deleted, etc.

Reply:
Maybe we should introduce two unhealthy states. One that means FreeIPA is not able to respond and another indicating it is still available.

Reply:
Okay that makes sense, I understand why you did it this way. 
Adding another unhealthy state might be ideal, something like being "unresponsive" vs being "unavailable", but that can definitely come with a future improvement.

Reply:
I filed CB-12245 to handle this case.